staging/src/k8s.io/client-go/restmapper/shortcut.go

```diff
@@ -145,16 +145,25 @@ func (e shortcutExpander) expandResourceShortcut(resource schema.GroupVersionRes
 			}
 		}
 
+		found := false
+		var rsc schema.GroupVersionResource
 		for _, item := range shortcutResources {
 			if len(resource.Group) != 0 && resource.Group != item.ShortForm.Group {
 				continue
 			}
 			if resource.Resource == item.ShortForm.Resource {
-				resource.Resource = item.LongForm.Resource
-				resource.Group = item.LongForm.Group
-				return resource
+				if found {
+					klog.V(1).Infof("Skipping ambiguous resource %q for %s. Continuing with %s. Please specify the fully qualified resource name.", item.LongForm, resource, rsc)
```

Comment:
a few thoughts:
1. This is a really long message that could be repeated multiple times... an example output when this happens would make this easier to see. It seems like some collapsing/combining should be done.
2. klog-formatted output doesn't seem like a great user experience (prefixing with shortcut.go / timestamp / etc)
3. adding logging into library methods rather than propagating the information back to the caller seems to go against the spirit of https://github.com/kubernetes/kubernetes/issues/94428

Reply:
as an alternative, NewShortcutExpander or an alternate constructor could take some input/indication of what to do with ambiguous resources

Reply:
@liggitt would you mind taking a look at the latest version of this PR?. Thanks.